mne/decoding/time_gen.py

```diff
@@ -320,14 +379,18 @@ def score(self, epochs=None, y=None):
                 raise RuntimeError('Please predict() epochs first or pass '
                                    'epochs to score()')
 
-        # clean gat.score() attributes
-        for att in ['scores_', 'scorer_', 'y_true_']:
-            if hasattr(self, att):
-                delattr(self, att)
-
         # Check scorer
-        # XXX Need API to identify proper scorer from the clf
-        self.scorer_ = accuracy_score if self.scorer is None else self.scorer
+        if self.scorer is not None:
+            self.scorer_ = self.scorer
+        else:
+            if is_classifier(self.clf) and self.predict_method == 'predict':
+                self.scorer_ = accuracy_score
+            elif is_regressor(self.clf) and self.predict_method == 'predict':
+                self.scorer_ = mean_squared_error
```

Comment:
There are def scorers only if `predict_method=='predict'` and we have a regressor or a classifier. If we have a different predict_method, or no classifier/regressor, we need to set the scorer.

(On the other hand, isn't every object we can use here either a regressor or a classifier?)